diesel/src/expression/expression_methods/sort_expression_methods.rs

```diff
@@ -0,0 +1,65 @@
+use expression::predicates::*;
+
+#[cfg(feature = "postgres")]
+pub trait Ordering : Sized {
+    /// Specify that nulls should come before other values in this ordering.
+    /// Normally, nulls come last when sorting in ascending order and first
+    /// when sorting in descending order.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("src/doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     users {
+    /// #         id -> Integer,
+    /// #         name -> VarChar,
+    /// #     }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// let query = users.select(id).order(name.asc().nulls_first());
+    /// assert_eq!("SELECT `users`.`id` FROM `users` ORDER BY `users`.`name` ASC NULLS FIRST",
+    ///            debug_sql!(query));
+    /// # }
+    /// ```
+    fn nulls_first(self) -> NullsFirst<Self> {
+        NullsFirst::new(self)
+    }
+    /// Specify that nulls should come after other values in this ordering.
+    /// Normally, nulls come last when sorting in ascending order and first
+    /// when sorting in descending order.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("src/doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     users {
+    /// #         id -> Integer,
+    /// #         name -> VarChar,
+    /// #     }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// let query = users.select(id).order(name.desc().nulls_last());
+    /// assert_eq!("SELECT `users`.`id` FROM `users` ORDER BY `users`.`name` DESC NULLS LAST",
+    ///            debug_sql!(query));
+    /// # }
+    /// ```
+    fn nulls_last(self) -> NullsLast<Self> {
+        NullsLast::new(self)
+    }
+}
+
+#[cfg(feature = "postgres")]
+impl<T> Ordering for Asc<T> {}
+
+#[cfg(feature = "postgres")]
+impl<T> Ordering for Desc<T> {}
```

Comment:
Should this be implemented for any `T: Expression`? It's completely valid to specify this without a defined order.